### PR TITLE
Early prototype

### PR DESCRIPTION
**Pull request by robinnedergard**
Weird type related bug required casting in function create. I see nothing wrong so don't know why type system is complaining.

### CODE REVIEW

File: Event/Settlement.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import * as isoly from "isoly"
+import { Base } from "./Base"
+
+export interface Settlement extends Base {
+	type: "settlement"
+	start: isoly.DateTime
+	end: isoly.DateTime
```
Comment (simonmika) [inline Event/Settlement.ts:7]: Not really related to the event, I guess it is the settlement period? Maybe this should be a part of the payout part?

File: Event/Settlement.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	currency: isoly.Currency
+	amountGross: number
```
Comment (simonmika) [inline Event/Settlement.ts:9]: Maybe shorten it to `amount`? Less descriptive though.

File: Event/Type.ts [hunk 3]
```diff
@@ -1,4 +1,15 @@
...
+	| "defer"
+	| "pend"
+	| "deny"
+	| "order"
+	| "cancel"
+	| "charge"
+	| "pay"
+	| "refund"
+	| "fail"
+	| "settlement"
```
Comment (simonmika) [inline Event/Type.ts:11]: Should be a verb. Maybe `settle`?

File: Event/Settlement.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import * as isoly from "isoly"
+import { Base } from "./Base"
+
+export interface Settlement extends Base {
```
Comment (simonmika) [inline Event/Settlement.ts:4]: Should be a verb like the other events. Maybe `Settle`?

File: Status.ts [hunk 5]
```diff
@@ -11,6 +11,7 @@ export type Status =
 	| "paid"
 	| "refunded"
 	| "synchronized"
+	| "settled"
```
Comment (simonmika) [inline Status.ts:14]: Use the same order as in events as far as possible.

File: Status.ts [hunk 6]
```diff
@@ -24,6 +25,7 @@ export namespace Status {
 		"paid",
 		"refunded",
 		"synchronized",
+		"settled",
```
Comment (simonmika) [inline Status.ts:28]: Order?

**Review by simonmika — APPROVED** [review]

**Review by simonmika — APPROVED** [review]

**Review by fredrikbryntesson — APPROVED** [review]

**Pull request merged**